### PR TITLE
remove PruneWhitelist, use prune-allowlist instead

### PR DESCRIPTION
**Pull request by pacoxu**
/kind cleanup
/kind deprecation
Follow up of  https://github.com/kubernetes/kubernetes/pull/117573 and https://github.com/kubernetes/kubernetes/pull/113116

```release-note
kubectl: remove deprecated flag prune-whitelist for apply, use flag prune-allowlist instead.
```

### CODE REVIEW

File: test/cmd/apply.sh [hunk 1]
```diff
@@ -231,15 +231,15 @@ __EOF__
...
 
-  ## kubectl apply --prune --prune-whitelist
+  ## kubectl apply --prune --prune-allowlist
   # Pre-Condition: no POD exists
   kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
   # apply pod a
   kubectl apply --prune -l prune-group=true -f hack/testdata/prune/a.yaml "${kube_flags[@]:?}"
   # check right pod exists
   kube::test::get_object_assert 'pods a' "{{${id_field:?}}}" 'a'
   # apply svc and don't prune pod a by overwriting whitelist
```
Comment (brianpursley) [inline test/cmd/apply.sh:241]:
```suggestion
  # apply svc and don't prune pod a by overwriting allowlist
```

File: test/cmd/apply.sh [hunk 2]
```diff
@@ -231,15 +231,15 @@ __EOF__
...
-  kubectl apply --prune -l prune-group=true -f hack/testdata/prune/svc.yaml --prune-whitelist core/v1/Service 2>&1 "${kube_flags[@]:?}"
+  kubectl apply --prune -l prune-group=true -f hack/testdata/prune/svc.yaml --prune-allowlist core/v1/Service 2>&1 "${kube_flags[@]:?}"
   kube::test::get_object_assert 'service prune-svc' "{{${id_field:?}}}" 'prune-svc'
   kube::test::get_object_assert 'pods a' "{{${id_field:?}}}" 'a'
   # apply svc and prune pod a with default whitelist
```
Comment (brianpursley) [inline test/cmd/apply.sh:245]:
```suggestion
  # apply svc and prune pod a with default allowlist
```

**Review by brianpursley — CHANGES_REQUESTED** [review]
Found a couple more places in the comments that I noticed, so we might as well change it everywhere to avoid any potential confusion in the future.

Otherwise, looks OK.

**Comment by ardaguclu** [comment]
This patch looks good to me. Could you please change the `whitelist` in here to `allowlist` https://github.com/kubernetes/kubernetes/blob/eb32969ab8f3d1455c33a595bba5b3ce4d8a03b8/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go#L169?

/approve
/triage accepted

File: test/cmd/apply.sh (hunk 1 above)
Comment (pacoxu) [reply test/cmd/apply.sh]: Updated.

**Comment by pacoxu** [comment]
> This patch looks good to me. Could you please change the `whitelist` in here to `allowlist`

Updated.

**Comment by miss15** [comment]
的

**Pull request merged**